src/test/java/io/axoniq/axonserver/connector/ResultStreamPublisherTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022. AxonIQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.axoniq.axonserver.connector;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.tck.PublisherVerification;
+import org.reactivestreams.tck.TestEnvironment;
+import org.testng.annotations.Test;
+
+/**
+ * Unit tests for {@link ResultStreamPublisher}.
+ *
+ * @author Sara Pellegrini
+ */
+public class ResultStreamPublisherTest extends PublisherVerification<Long> {
+
+    private final TestEnvironment env;
+
+    public ResultStreamPublisherTest() {
+        this(new TestEnvironment(1_000));
+    }
+
+    private ResultStreamPublisherTest(TestEnvironment env) {
+        super(env);
+        this.env = env;
+    }
+
+    @Override
+    public Publisher<Long> createPublisher(long elements) {
+        return new ResultStreamPublisher<>(() -> new FakeResultStream(elements));
+    }
+
+    @Override
+    public Publisher<Long> createFailedPublisher() {
+        return new ResultStreamPublisher<>(FailedResultStream::new);
+    }
+
+    @Test
+    public void required_spec317_mustSupportACumulativePendingElementCountGreaterThenLongMaxValue() throws Throwable {
```

Comment:
As this tests a required specification (number 317, apparently), would you mind sharing a link to the given spec as documentation to this test?
I am not directly assuming we need this, but it would help future readers of this test class to better understand (and find) the specification in question.

Or, is it perhaps an override from the `PublisherVerification`?

Reply:
I've added the link to the specification. 
This is an additional test that was not originally present in PublisherVerification.